### ISSUE TITLE
Add ARIA roles and attributes to account registration page

### ISSUE THREAD

**Issue by jimchamp** [internetarchive/openlibrary#10801]
### Problem

Dynamic sections of the account registration form lack ARIA roles and attributes, likely hindering navigation for patrons using screen readers.

### Reproducing the bug

1. Go to ...
2. Do ...

* Expected behavior:
* Actual behavior:

### Context

- Browser (Chrome, Safari, Firefox, etc):
- OS (Windows, Mac, etc):
- Logged in (Y/N):
- Environment (prod, dev, local): prod

### Breakdown

@jimchamp to provide a breakdown of requirements

#### Requirements Checklist
* [ ]

#### Related files

*

#### Stakeholders

*
<hr>

#### Instructions for Contributors

- Please [run these commands](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#working-on-your-branch) to ensure your repository is up to date **before** [creating a new branch](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) to work on this issue and **each time after** pushing code to Github, because the pre-commit bot may add commits to your PRs upstream.

**Comment by mystic-06** [comment]
Hey, I'd like to work on this issue, is it still available?

**Comment by jimchamp** [comment]
Sorry for the delay -- it is still available.

We have recently added a [checkbox and select element](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/account/create.html#L82-L94) to our [registration page](https://openlibrary.org/account/create).  Whenever clicking the checkbox will toggle the `<select>` element's visibility.

<img width="537" height="210" alt="Image" src="https://github.com/user-attachments/assets/bb2ef264-c262-4174-b171-7b6051705192" />

These elements need to be updated to include [ARIA roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles) to better communicate what is happening in the UI for folks that use screen readers.

Please let me know if you need more details.
